### PR TITLE
Analytics query module

### PR DESCRIPTION
**Pull request by ktuite**
This Analytics query module calculates different metrics about server usage (anonymous averages and aggregates only). Admins can preview these metrics through the `/v1/analytics/preview` API resource. Frontend uses (will soon use) this endpoint to render a nice summary of these metrics.

Notes about what to change when the metrics need to be updated, along with a helper script that parses the analytics form XML, can be found here: https://gist.github.com/ktuite/931d75b3a5e7f14204949bf1f40f3426

Screen Shot 2021-09-18 at 1 27 02 PM

### CODE REVIEW

File: lib/model/query/analytics.js [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+  return projArray;
+}));
+
+const previewMetrics = () => (({ Analytics }) => Promise.all([
+  Analytics.databaseSize(),
+  Analytics.backupsEnabled(),
+  Analytics.encryptedProjects(),
+  Analytics.biggestForm(),
+  Analytics.countAdmins(),
+  Analytics.auditLogs(),
```
Comment (ktuite) [inline lib/model/query/analytics.js:412]: maybe put in `Analytics.systemMetrics()`

File: lib/model/query/analytics.js [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+    metrics.system.num_admins[key] = value;
+  for (const [key, value] of Object.entries(audits))
+    metrics.system.num_audit_log_entries[key] = value;
+  metrics.projects = projMetrics;
+
+
+  return metrics;
+}));
+
+module.exports = {
```
Comment (ktuite) [inline lib/model/query/analytics.js:434]: export one per row/line or thematically grouped

File: test/integration/other/analytics-queries.js [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+// Utility
+const simpleInstance = (newInstanceId) => testData.instances.simple.one
+  .replace('one</instance', `${newInstanceId}</instance`);
+
+const withSimpleIds = (deprecatedId, instanceId) => testData.instances.simple.one
+  .replace('one</instance', `${instanceId}</instanceID><deprecatedID>${deprecatedId}</deprecated`);
+
+describe.only('analytics task queries', () => {
+  describe('general server metrics', () => {
+    it('should count audit log entries', testContainer( async (container) => {
```
Comment (ktuite) [inline test/integration/other/analytics-queries.js:19]: remove space before `async`

File: test/data/analytics.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+const appRoot = require('app-root-path');
+const { sql } = require('slonik');
+const { Actor, Form, Project, Submission, User } = require(appRoot + '/lib/model/frames');
+const { forms, instances } = require('./xml');
+const { createReadStream, readFileSync } = require('fs');
+
+const createTestUser = async (container, name, role, projectId, recent = true) => {
```
Comment (ktuite) [inline test/data/analytics.js:7]: move these helper functions into the analytics test file itself

Comment (ktuite) [reply test/data/analytics.js:7]: use the api to do this

**Review by issa-tseng — APPROVED** [review]

**Review by ktuite — COMMENTED** [review]
notes from interactive review with @issa-tseng

**Pull request merged**